fdbserver/workloads/WriteTagThrottling.actor.cpp

```diff
@@ -0,0 +1,271 @@
+		tpsRate = getOption(options, LiteralStringRef("tpsRate"), 300.0);
+		numWritePerTx = getOption(options, LiteralStringRef("numWritePerTx"), 1);
+		numReadPerTx = getOption(options, LiteralStringRef("numReadPerTx"), 1);
+		numClearPerTx = getOption(options, LiteralStringRef("numClearPerTx"), 1);
+
+		writeThrottle = getOption(options, LiteralStringRef("writeThrottle"), false);
+		badActorPerClient = getOption(options, LiteralStringRef("badActorPerClient"), 1);
+		goodActorPerClient = getOption(options, LiteralStringRef("goodActorPerClient"), 1);
+		actorCount = goodActorPerClient + badActorPerClient;
+
+		keyCount = getOption(options, LiteralStringRef("keyCount"),
+		                     std::max(3000, actorCount * 3)); // enough keys to avoid too many conflicts
+		txInterval = actorCount * 1.0 / getOption(options, LiteralStringRef("txPerSecond"), 1000);
+
+		badReadTag = TransactionTag(std::string("badReadTag"));
+		badWriteTag = TransactionTag(std::string("badWriteTag"));
+		goodTag = TransactionTag(std::string("goodTag"));
+	}
+
+	virtual std::string description() { return WriteTagThrottlingWorkload::NAME; }
+
+	// choose a tag
+	ACTOR static Future<Void> _setup(Database cx, WriteTagThrottlingWorkload* self) {
+		state TransactionPriority priority = deterministicRandom()->randomChoice(allTransactionPriorities);
+		state TagSet tagSet1;
+		state TagSet tagSet2;
+		state TagSet tagSet3;
+		tagSet1.addTag(self->badWriteTag);
+		tagSet2.addTag(self->badReadTag);
+		tagSet3.addTag(self->goodTag);
+
+		wait(ThrottleApi::throttleTags(cx, tagSet1, self->tpsRate, self->throttleDuration, TagThrottleType::MANUAL,
+		                               priority));
+		wait(ThrottleApi::throttleTags(cx, tagSet2, self->tpsRate, self->throttleDuration, TagThrottleType::MANUAL,
+		                               priority));
+		wait(ThrottleApi::throttleTags(cx, tagSet3, self->tpsRate, self->throttleDuration, TagThrottleType::MANUAL,
+		                               priority));
+
+		return Void();
+	}
+	Future<Void> setup(const Database& cx) override { return clientId ? Void() : _setup(cx, this); }
+	ACTOR static Future<Void> _start(Database cx, WriteTagThrottlingWorkload* self) {
+		vector<Future<Void>> clientActors;
+		int actorId;
+		for (actorId = 0; actorId < self->goodActorPerClient; ++actorId) {
+			clientActors.push_back(clientActor(false, actorId, 0, cx, self));
+		}
+		for (actorId = 0; actorId < self->badActorPerClient; ++actorId) {
+			clientActors.push_back(clientActor(true, actorId, self->badOpRate, cx, self));
+		}
+		wait(timeout(waitForAll(clientActors), self->testDuration, Void()));
+		return Void();
+	}
+	virtual Future<Void> start(Database const& cx) { return _start(cx, this); }
+	virtual Future<bool> check(Database const& cx) {
+		if (badActorTxNum == 0 && goodActorTxNum == 0) {
+			TraceEvent(SevError, "NoTransactionsCommitted");
+			return false;
+		}
+		// TODO check whether write throttling work correctly after enabling that feature
```

Comment:
Basically this workload is not used to test correctness in simulation. 
Having the test txt file will waste correctness tests resource. 

I'd like to suggest: (1) Remove the  tests/WriteTagThrottling.txt  until the workload check is real; (2) add a line `if (g_network->isSimulated() ) {
// Add a TraceEvent(SevError) and an ASSERT(false) so that whoever uses this workload in simulation will get noticed. 
}`

Reply:
I prefer option 2. Do you mean adding the line in `setup`?

Reply:
It can be in setup or check. it does not matter that much. After you adding (2), correctness will fail if you don't add (1). Give it a try. :)

Reply:
I just remembered one thing. 
I still would like to make this workload available for simulation. The reason is though I don't check write latency and the number of Retries.throttle here (because it's hard to get a clear threshold), my simulation test still print above metrics which show throttling work.

Reply:
I added a comment below.  if the tag throttling does not work, will the test fail in simulation?

Reply:
I can see what you mean. Some simplest checks can be added to check whether tag throttling works. I updated it. : )